service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/batchpool/CreateAzureBatchPoolStep.java

```diff
@@ -75,11 +76,27 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     logger.info(
         String.format(PET_UAMI_FOUND, GetManagedIdentityStep.getManagedIdentityName(context)));
 
-    final BatchPoolUserAssignedManagedIdentity userAssignedManagedIdentity =
+    List<BatchPoolUserAssignedManagedIdentity> uamiForBatchPool = new ArrayList<>();
+
+    // This UAMI is required. It is the user's pet and allows them to run compute.
+    uamiForBatchPool.add(
         new BatchPoolUserAssignedManagedIdentity(
             azureCloudContext.getAzureResourceGroupId(),
             GetManagedIdentityStep.getManagedIdentityName(context),
-            null);
+            null));
+
+    String maybeActionIdentity =
+        context.getWorkingMap().get(GetActionManagedIdentityStep.ACTION_IDENTITY, String.class);
+    if (maybeActionIdentity != null) {
+      // This UAMI may/may not exist (and that's OK). It allows the user to pull images from a
+      // private azure container repository.
+      logger.info("Assigning action managed identity {} to batch pool", maybeActionIdentity);
+      uamiForBatchPool.add(
+          new BatchPoolUserAssignedManagedIdentity(
+              azureCloudContext.getAzureResourceGroupId(), maybeActionIdentity, null));
+    } else {
+      logger.info("Skipping assigning any action managed identities to batch pool.");
```

Comment:
Include the workspace ID and resource ID in this log statement for better debug- and audit-ability.